packages/sso-menu/src/SSOMenu.tsx

```diff
@@ -0,0 +1,248 @@
+`;
+
+const accountMenuGroupStyle = css`
+  padding: 20px 20px 14px;
+`;
+
+const accountButtonStyle = css`
+  margin-top: 12px;
+  width: 100%;
+  display: inline-flex;
+  justify-content: center;
+`;
+
+const menuItemTextStyle = css`
+  font-size: 14px;
+  line-height: 16px;
+  cursor: pointer;
+  margin: 0px;
+`;
+
+const descriptionStyle = css`
+  margin: 0px;
+  display: block;
+  font-weight: normal;
+  font-size: 12px;
+  color: ${uiColors.gray.dark2};
+  text-decoration: none;
+`;
+
+const menuItemPadding = css`
+  padding-left: 20px;
+`;
+
+const logoutContainerHeight = css`
+  height: 46px;
+`;
+
+const menuItems = [
+  {
+    displayName: 'Atlas',
+    description: 'cloud.mongodb.com',
+    href: 'https://cloud.mongodb.com',
+    slug: 'atlas',
+  },
+  {
+    displayName: 'University',
+    description: 'university.mongodb.com',
+    href: 'https://university.mongodb.com',
+    slug: 'university',
+  },
+  {
+    displayName: 'Cloud Support',
+    description: 'support.mongodb.com',
+    href: 'https://support.mongodb.com',
+    slug: 'support',
+  },
+] as const;
+
+const slugs = menuItems.map(mi => mi.slug);
+
+export const ActiveProduct = {
+  Atlas: 'atlas',
+  University: 'university',
+  Support: 'support',
+} as const;
+
+type ActiveProduct = typeof ActiveProduct[keyof typeof ActiveProduct];
+
+interface SSOMenuProps {
+  /**
+   * Object that contains information about the active user. {name: 'string', email: 'string'}
+   */
+  user: { name: string; email: string };
+
+  /**
+   * MongoDB product that is currently active: ['atlas', 'university', 'support'].
+   */
+  activeProduct: ActiveProduct;
+
+  /**
+   * Callback invoked when user logs out.
+   */
+  onLogout?: React.MouseEventHandler;
+
+  /**
+   * Callback invoked when user switches products.
+   */
+  onProductChange?: React.MouseEventHandler;
+}
+
+/**
+ * # SSOMenu
+ *
+ * ```
+<SSOMenu
+    user={{ name: 'Alex Smith', email: 'alex.smith@youwork.com' }}
+    activeProduct="atlas"
+    onLogout={() => console.log('On logout')}
+    onProductChange={() => console.log('Switching products')}
+  />
+ * ```
+ * @param props.user Object that contains information about the active user. {name: 'string', email: 'string'}
+ * @param props.activeProduct  MongoDB product that is currently active: ['atlas', 'university', 'support'].
+ * @param props.onLogout Callback invoked when user logs out.
+ * @param props.onProductChange Callback invoked when user switches products.
+ *
+ */
+function SSOMenu({
+  user: { name, email },
+  activeProduct,
+  onLogout = () => {},
+  onProductChange = () => {},
+}: SSOMenuProps) {
+  const [open, setOpen] = useState(false);
+  return (
+    <button
+      onClick={() => setOpen(curr => !curr)}
+      className={cx(menuButtonStyle, {
+        [activeMenuButtonStyle]: open,
+      })}
+    >
+      <span className={menuNameStyle}>{name}</span>
+      {open ? (
+        <Icon glyph="CaretUp" fill={uiColors.white} className={iconStyle} />
```

Comment:
Wouldn't it be better to implement this as a single icon so that the color can transition?

Reply:
👍